src/mcts/search.cc

```diff
@@ -838,6 +844,19 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
 
   // Fetch the current best root node visits for possible smart pruning.
   const int64_t best_node_n = search_->current_best_edge_.GetN();
+  // and Q for moves_left logic.
+  const float best_node_q = search_->current_best_edge_.GetQ(0);
+
+  bool use_moves_left = moves_left_support_ && params_.GetUseMovesLeft();
+  int moves_left = 0;
+  float moves_threshold = params_.GetMovesLeftThreshold();
+  float best_node_m = search_->current_best_edge_.GetM(0.0f);
+
+  if (best_node_q > moves_threshold) {
+    moves_left = -1;
```

Comment:
Something tells me that this variable name is also poor. How can number of moves left be negative?
Also something about the variable that can be 0, -1 or 1...
Maybe either enum.. `{kNoLengthAdjustment, kTryLengthen, kTryShorten}`
or two bools, something lie `gamelength_penalty_enabled` `want_longer_game`..

Reply:
Renamed to `moves_left_sign` and used as a sign in the score calculation.